blueprints/flexible-topologies.md

```diff
@@ -250,11 +250,29 @@ response JSON:
 
 ##### `/deliveryservices` endpoints
 
-All relevant Delivery Service APIs will have their JSON request and response objects modified to include a new `topology` field which references the name of the topology it's assigned to:
-```JSON
+All relevant Delivery Service APIs will have their JSON request and response objects modified to include the following new fields:
+- `topology` - the name of the topology it's assigned to
+- `firstHeaderRewrite` - the header_rewrite ATS config that will be applied to the *first tier* caches in the Delivery Service's Topology
+- `middleHeaderRewrite` - the header_rewrite ATS config that will be applied to all *middle tier* caches in this Delivery Service's Topology
+- `lastHeaderRewrite` - the header_rewrite ATS config that will be applied to all *last tier* caches in this Delivery Service's Topology
+
+**Note:** these three header_rewrite "buckets" were chosen because header_rewrite configs usually fall into one of three categories:
+- affects requests to/from the client ("first")
+- affects requests to/from the origin ("last")
+- don't affect clients or origins ("middle")
+
+**API Constraints:**
+- `firstHeaderRewrite`, `middleHeaderRewrite`, and `lastHeaderRewrite` can only be set if `topology` is not null
+- `edgeHeaderRewrite` and `midHeaderRewrite` cannot be set if `topology` is not null
```

Comment:
Instead of having parallel mutually exclusive fields serving the same purpose, couldn't we just drop the first and last fields?  Seems like that creates additional tradition steps when converting to topologies.  The concept of a middleHeaderRewrite doesn't currently exist, so it's fine to be topology only.

Reply:
This is on purpose -- we want it to be part of the process of migrating legacy DSes to Topologies. Part of that is making sure the existing header-rewrite configs are safe and make sense in the Topology you're assigning the DS to.